lib/earmark_parser/line_scanner.ex

```diff
@@ -58,13 +56,49 @@ defmodule EarmarkParser.LineScanner do
 
   def scan_lines(lines, options, recursive) do
     lines_with_count(lines, options.line - 1)
-    |> get_mapper(options).(fn line -> type_of(line, options, recursive) end)
+    |> with_lookahead(options, recursive)
   end
 
   defp lines_with_count(lines, offset) do
     Enum.zip(lines, offset..(offset + Enum.count(lines)))
   end
 
+  defp with_lookahead([line_lnb | lines], options, recursive) do
+    case type_of(line_lnb, options, recursive) do
+      %Line.Fence{delimiter: delimiter, indent: indent} = fence ->
+        stop =
+          # We should stop on another code block, any close html tag,
+          # or on less indent if there is any indentation.
+          case indent do
+            0 ->
+              ~r/\A(\s*)(#{delimiter})\s*([^`\s]*)\s*\z|\A(\s{0,3})<\/([-\w]+?)>/u
+
+            _ ->
+              ~r/\A(\s*)(#{delimiter})\s*([^`\s]*)\s*\z|\A(\s{0,3})<\/([-\w]+?)>|\A\s{#{indent - 1}}.+\z/u
```

Comment:
About the last part `\A\s{#{indent - 1}}.+\z`, since there is `.+` it would also match longer indents, or did I miss it?

Reply:
It would, you are correct. I will add a test and address it.